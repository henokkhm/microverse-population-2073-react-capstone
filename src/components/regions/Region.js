import PropTypes from 'prop-types';

import styles from '../../styles/Region.module.css';
import rightArrow from '../../assets/icons/circled-arrow.svg';

function Region({ name, populationIn2073, index }) {
  const isBackgroundDark = index % 4 === 1 || index % 4 === 2;
  const bgStyle = {
    backgroundColor: isBackgroundDark
      ? 'var(--primary-color-200)'
      : 'var(--primary-color-300)',
  };

  return (
    <li className={styles.region} style={bgStyle}>
      <button className={styles.arrowBtn} type="button">
        <img
          className={styles.arrowBtnIcon}
          src={rightArrow}
          alt="circled arrow pointing right"
        />
      </button>
      <div className={styles.regionContent}>
        <span className={styles.regionName}>{name}</span>
        <span className={styles.regionPopulationIn2073}>
          {populationIn2073.toLocaleString()}
        </span>
      </div>
    </li>
  );
}

Region.propTypes = {
  name: PropTypes.string.isRequired,
  populationIn2073: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
};

export default Region;
