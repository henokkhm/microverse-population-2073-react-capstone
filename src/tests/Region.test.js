import { render, screen } from '@testing-library/react';

import Region from '../components/regions/Region';

describe('The Region component', () => {
  it('renders region name and population', () => {
    const { name, imageURL, populationIn2073 } = {
      name: 'Asia [mock]',
      imageURL: '/images/asia.png',
      populationIn2073: 5174232028,
    };

    render(
      <Region
        name={name}
        imageURL={imageURL}
        populationIn2073={populationIn2073}
        index={0}
      />,
    );

    expect(screen.getByText(/Asia/i)).toBeInTheDocument();
    expect(screen.getByText(/5,174,232,028/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { name, imageURL, populationIn2073 } = {
      name: 'Asia [mock]',
      imageURL: '/images/asia.png',
      populationIn2073: 5174232028,
    };

    const { asFragment } = render(
      <Region
        name={name}
        imageURL={imageURL}
        populationIn2073={populationIn2073}
        index={0}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
