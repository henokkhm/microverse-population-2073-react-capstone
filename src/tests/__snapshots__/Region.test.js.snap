// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Region component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="region"
  >
    <button
      class="arrowBtn"
      type="button"
    >
      <img
        alt="circled arrow pointing right"
        class="arrowBtnIcon"
        src="circled-arrow.svg"
      />
    </button>
    <div
      class="regionContent"
    >
      <img
        alt="map of Asia [mock] region"
        class="regionMap"
        src="/images/asia.png"
      />
      <span
        class="regionName"
      >
        Asia [mock]
      </span>
      <span
        class="regionPopulationIn2073"
      >
        5,174,232,028 people
      </span>
    </div>
  </li>
</DocumentFragment>
`;
